app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala

```diff
@@ -16,22 +16,98 @@
-import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.helptosavestridefrontend.connectors.HelpToSaveConnector
+import uk.gov.hmrc.helptosavestridefrontend.forms.{GiveNINOForm, NINOValidation}
+import uk.gov.hmrc.helptosavestridefrontend.models.PayePersonalDetails
+import uk.gov.hmrc.helptosavestridefrontend.models.eligibility.EligibilityCheckResult
+import uk.gov.hmrc.helptosavestridefrontend.models.eligibility.EligibilityCheckResult.{AlreadyHasAccount, Eligible, Ineligible}
+import uk.gov.hmrc.helptosavestridefrontend.util.{Logging, NINOLogMessageTransformer, base64Encode, toFuture}
 import uk.gov.hmrc.helptosavestridefrontend.views
+import uk.gov.hmrc.http.HeaderCarrier
 
-class StrideController @Inject() (val authConnector: AuthConnector,
-                                  val messagesApi:   MessagesApi)(implicit val frontendAppConfig: FrontendAppConfig)
-  extends FrontendController with StrideAuth with I18nSupport {
+import scala.concurrent.Future
 
-  val getStartPage: Action[AnyContent] = authorisedFromStride{ implicit request ⇒
-    Ok(views.html.start_page())
-  }(routes.StrideController.getStartPage())
+class StrideController @Inject() (val authConnector:       AuthConnector,
+                                  val helpToSaveConnector: HelpToSaveConnector,
+                                  val frontendAppConfig:   FrontendAppConfig,
+                                  messageApi:              MessagesApi)(implicit val ninoValidation: NINOValidation,
+                                                                        transformer: NINOLogMessageTransformer)
+  extends StrideFrontendController(messageApi, frontendAppConfig) with StrideAuth with I18nSupport with Logging {
 
+  def getEligibilityPage: Action[AnyContent] = authorisedFromStride { implicit request ⇒
+    Ok(views.html.get_eligibility_page(GiveNINOForm.giveNinoForm))
+  }(routes.StrideController.getEligibilityPage())
+
+  def checkEligibilityAndGetPersonalInfo: Action[AnyContent] = authorisedFromStride { implicit request ⇒
+    GiveNINOForm.giveNinoForm.bindFromRequest().fold(
+      withErrors ⇒ Ok(views.html.get_eligibility_page(withErrors)),
+      form ⇒ {
+        val ninoEncoded = new String(base64Encode(form.nino))
+
+        val r = for {
+          eligibility ← helpToSaveConnector.getEligibility(ninoEncoded)
+          personalDetails ← getPersonalDetails(eligibility, ninoEncoded)
+        } yield eligibility -> personalDetails
+
+        r.fold(
+          error ⇒ {
+            logger.warn(s"error during get eligibility result and pay-personal-info, error: $error")
+            internalServerError()
+          },
+          {
+            case (Eligible(_), Some(details)) ⇒
+              Ok(views.html.you_are_eligible(details))
+            case (Eligible(_), None) ⇒
+              logger.warn("user is eligible but could not retrieve pay-personal-info")
+              SeeOther(routes.StrideController.noPayeDetailsFound().url)
```

Comment:
this case will never happen as the code stands now - we might not need a special page for this case